src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,105 @@
 package core.basesyntax;
 
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+
 public class SalaryInfo {
+    private static final SimpleDateFormat DATE_FORMAT = new SimpleDateFormat("dd.MM.yyyy");
+    private static final int DATE_PART = 0;
+    private static final int NAME_PART = 1;
+    private static final int HOURS_PART = 2;
+    private static final int INCOMING_PER_PART = 3;
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        Map<String, Integer> mapOfName = putNameInMap(names);
```

Comment:
The implementation could be so much simpler 


```
public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
        StringBuilder report = new StringBuilder("Report for period ");
        report.append(getFormattedData(dateFrom, "-", dateTo));

        for (String name : names) {
            int totalSalary = 0;
            for (String values : data) {
                String[] dataInfo = values.split(" ");
                if (isInRangeOfPeriod(dataInfo[0], dateFrom, dateTo)
                        && name.equals(dataInfo[1])) {
                    totalSalary += Integer.parseInt(dataInfo[2]) * Integer.parseInt(dataInfo[3]);
                }
            }
            report.append(System.lineSeparator())
                    .append(getFormattedData(name, "-", Integer.toString(totalSalary)));
        }
        return report.toString();
    }

    private boolean isInRangeOfPeriod(String data, String dataFrom, String dataTo) {
        return !encodeData(data).isBefore(encodeData(dataFrom))
                && !encodeData(data).isAfter(encodeData(dataTo));
    }

    private LocalDate encodeData(String data) {
        return LocalDate.parse(data, DATE_FORMATTER);
    }

    private String getFormattedData(String...data) {
        return String.join(" ", data);
    }
```

Reply:
Very grateful for your helping